### DISCUSSION TITLE
the incident direction in intersection information is different from the defined ray direction

### DISCUSSION THREAD

**Discussion by boringfish** [mitsuba-renderer/mitsuba3#1356] [Q&A] [answered]
Hi，everyone. I found a strange problem, and not sure if it is a bug.
I want to trace a ray and get intersection information. The ray is defined with an origin poistion and a ray direction. 
In general, the incident direction (w_i) in the returned intersection information should be opposite to the ray direction defined in the tracing ray. But I find that sometimes  (w_i) is different from the defined ray direction.
A simple test code is as follows:
```
scene = mi.load_file(
        "C:/Users/China/Downloads/mitsuba-data-master/mitsuba-data-master/docs/scenes/medium_testheterogeneous_sss.xml")
sampler = mi.load_dict({'type': 'independent', 'sample_count': 16})
ray = mi.Ray3f(mi.Point3f(test[6]), mi.Vector3f([0.45039022, -0.8601879,  0.23921824]))
si = scene.ray_intersect(ray)
print(si)
```
and the output is:
```
SurfaceInteraction[
  t = [0.60113],
  time = [0],
  wavelengths = [],
  p = [[0.483971, 0.388641, 0.272843]],
  shape = [0x20c48c1c230],
  uv = [[0.501742, 0.621071]],
  n = [[-0.0147006, 0.635613, -0.771868]],
  sh_frame = Frame[
    s = [[0.873842, -0.343769, -0.343837]],
    t = [[-0.484143, -0.680354, -0.550204]],
    n = [[-0.0447877, 0.647257, -0.760955]]
  ],
  dp_du = [[1.64256, -0.686431, -0.596541]],
  dp_dv = [[-0.496472, -1.29662, -1.05828]],
  dn_du = [[0, 0, 0]],
  dn_dv = [[0, 0, 0]],
  duv_dx = [[0, 0]],
  duv_dy = [[0, 0]],
  wi = [[-0.607023, -0.23556, 0.758969]],
  prim_index = [50954],
  instance = [0x0]
]
```
One can be observed is that  wi = [[-0.607023, -0.23556, 0.758969]] is very different from  the ray direction [0.45039022, -0.8601879,  0.23921824]. Is there any problems, or do I misunderstand something?

**Comment by xacond00** [comment]
Wi is the direction in the local frame, that's why it is different from ray direction.

**Comment by ziyi-zhang** [comment, marked as answer]
Hi,
As @xacond00 mentioned, it’s defined in the local frame built with the shading normal. Additionally, it is pointing outward. You can find more details in “interaction.h”.
